src/components/forms/GPUDeviceForm.tsx

```diff
@@ -0,0 +1,256 @@
+import { FC } from "react";
+import {
+  Button,
+  Icon,
+  Input,
+  Select,
+  Tooltip,
+  useNotify,
+} from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { LxdGPUDevice } from "types/device";
+import { InstanceAndProfileFormikProps } from "./instanceAndProfileFormValues";
+import ScrollableConfigurationTable from "components/forms/ScrollableConfigurationTable";
+import { fetchProfiles } from "api/profiles";
+import { EditInstanceFormValues } from "pages/instances/EditInstance";
+import { getConfigurationRowBase } from "components/ConfigurationRow";
+import { getInheritedGPUs } from "util/configInheritance";
+import { fetchConfigOptions } from "api/server";
+import { useSupportedFeatures } from "context/useSupportedFeatures";
+import { toConfigFields } from "util/config";
+import ConfigFieldDescription from "pages/settings/ConfigFieldDescription";
+import Loader from "components/Loader";
+
+interface Props {
+  formik: InstanceAndProfileFormikProps;
+  project: string;
+}
+
+const GPUDevicesForm: FC<Props> = ({ formik, project }) => {
```

Comment:
a bit of a general question, gpu devices existed before the `metadata/configuration` endpoint. Are we going to support GPU devices in the UI for older lxd versions?